portfolio/src/main/webapp/js/background.js

```diff
@@ -0,0 +1,291 @@
+/**
+ * @fileoverview Creates a live visualization of balls that repel each other
+ * and bounce off of the walls.
+ */
+const HEIGHT_TO_WIDTH = 0.3;
+const EFFECT_RADIUS_TO_WIDTH = 0.1;
+const BALL_SIZE_TO_WIDTH = 0.01;
+const NUM_BALLS = 25;
+const INITIAL_BALL_VELOCITY_MAGNITUDE = 1;
+/**
+ * Runs initial set up of the background and background canvas.
+ */
+function setup() {
+  const backgroundContainer = document.getElementById("background-container");
+
+  const backgroundCanvas =
+    createCanvas(backgroundContainer.clientWidth,
+      backgroundContainer.clientWidth * HEIGHT_TO_WIDTH);
+  backgroundCanvas.parent("background-container");
+
+  frameRate(60);
+
+  initTank();
+}
+
+let ballTank;
+/**
+ * Initializes a tank for the background.
+ */
+function initTank() {
+  const ballSize = BALL_SIZE_TO_WIDTH * width;
+  ballTank = new Tank(width, height, EFFECT_RADIUS_TO_WIDTH * width, ballSize);
+
+  // Create balls and add them to the tank.
+  // Each start with random positions and velocities.
+  for (let i = 0; i < NUM_BALLS; i++) {
+    const randomPos = createVector(Math.random() * ballTank.width,
+      Math.random() * ballTank.height);
+
+    // Gets a random vector on the unit circle as to randomize 
+    // the initial velocity direction.
+    const randomVelocity = p5.Vector.random2D();
+    randomVelocity.mult(INITIAL_BALL_VELOCITY_MAGNITUDE);
+
+    const newBall = new Ball(randomPos, randomVelocity);
+    ballTank.addBall(newBall);
+  }
+}
+
+/**
+ * Resizes the canvas when the window is resized.
+ */
+function windowResized() {
```

Comment:
Maybe I missed something, is this function used anywhere?

Reply:
This and the draw function referenced below are called by p5.js, the graphics framework that I'm using here. Any thoughts on what I should include in the comments to make this more clear?

Reply:
Ah that makes sense. As someone unfamiliar with that library, yeah it would help for me if each of the "overridden" functions linked to the p5 docs like https://p5js.org/reference/#/p5/windowResized here.

Reply:
That makes sense. Ill do that now.

Reply:
Cool, thanks! Just went back and read a bit about JSDoc, seems there's an @implements tag as well. Could use that https://jsdoc.app/tags-implements.html. Up to you.